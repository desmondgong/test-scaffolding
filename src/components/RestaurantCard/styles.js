const styles = (theme) => ({
  root: {
    width: 280,
    margin: theme.spacing(2),
    boxShadow: '1px 1px 5px #dde1f8',
  },
  media: {
    height: 120,
  },
});
export default styles;
