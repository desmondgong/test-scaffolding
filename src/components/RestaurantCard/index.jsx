import React from 'react';
import PropTypes from 'prop-types';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActionArea from '@material-ui/core/CardActionArea';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import styles from './styles';

const useStyles = makeStyles(styles);
const RestaurantCard = (props) => {
  const { name, thumb, address, url } = props;
  const classes = useStyles(props);
  return (
    <Card className={classes.root}>
      <CardActionArea
        onClick={() => {
          window.open(url, '_blank');
        }}
      >
        {thumb && <CardMedia className={classes.media} image={thumb} title={name} />}
        <CardContent>
          <Typography variant="body1">{name}</Typography>
          <Typography variant="body2" color="textSecondary">
            {address}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

RestaurantCard.propTypes = {
  name: PropTypes.string.isRequired,
  thumb: PropTypes.string,
  address: PropTypes.string,
  url: PropTypes.string,
};
RestaurantCard.defaultProps = {
  thumb: '',
  address: '',
  url: '',
};

export default RestaurantCard;
