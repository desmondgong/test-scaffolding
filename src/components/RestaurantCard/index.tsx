import React, { ReactElement } from 'react';
import Card from '@material-ui/core/Card';
import CardMedia from '@material-ui/core/CardMedia';
import CardContent from '@material-ui/core/CardContent';
import CardActionArea from '@material-ui/core/CardActionArea';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';
import { Restaurant } from '../../types';
import styles from './styles';

export interface Props extends Restaurant {}

const useStyles = makeStyles(styles);

const RestaurantCard = (props: Props): ReactElement => {
  const { name, thumb = '', address = '', url = '' } = props;
  const classes = useStyles(props);
  return (
    <Card className={classes.root}>
      <CardActionArea
        onClick={() => {
          window.open(url, '_blank');
        }}
      >
        {thumb && <CardMedia className={classes.media} image={thumb} title={name} />}
        <CardContent>
          <Typography variant="body1">{name}</Typography>
          <Typography variant="body2" color="textSecondary">
            {address}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
};

RestaurantCard.defaultProps = {
  thumb: '',
  address: '',
  url: '',
};

export default RestaurantCard;
