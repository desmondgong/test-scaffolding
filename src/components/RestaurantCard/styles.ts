import { Theme } from '@material-ui/core';
import { BaseCSSProperties } from '@material-ui/styles';

export interface StyleProps {
  root: BaseCSSProperties;
  media: BaseCSSProperties;
}
const styles = (theme: Theme): StyleProps => ({
  root: {
    width: 280,
    margin: theme.spacing(2),
    boxShadow: '1px 1px 5px #dde1f8',
  },
  media: {
    height: 120,
  },
});
export default styles;
