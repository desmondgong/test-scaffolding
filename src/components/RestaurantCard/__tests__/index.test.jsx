import React from 'react';
import renderer from 'react-test-renderer';
import { renderWithTheme } from '../../../tests/testUtils';
import RestaurantCard from '..';

const props = {
  name: 'Kitchen',
  thumb: 'https://www.zomato.com/img',
  address: 'Ponsonby, Auckland',
  url: 'https://www.zomato.com/',
};

it('Matches the snapshots', () => {
  const component = renderer.create(<RestaurantCard {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Renders correctly', () => {
  const { getByText } = renderWithTheme(<RestaurantCard {...props} />);

  expect(getByText(props.name)).toBeInTheDocument();
});
