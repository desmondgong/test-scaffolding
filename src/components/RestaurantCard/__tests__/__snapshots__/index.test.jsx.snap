// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches the snapshots 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <div
      className="MuiCardMedia-root makeStyles-media-2"
      style={
        Object {
          "backgroundImage": "url(\\"https://www.zomato.com/img\\")",
        }
      }
      title="Kitchen"
    />
    <div
      className="MuiCardContent-root"
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Kitchen
      </p>
      <p
        className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
      >
        Ponsonby, Auckland
      </p>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight"
    />
  </button>
</div>
`;
